lms/djangoapps/program_enrollments/api/v1/tests/test_views.py

```diff
@@ -425,12 +412,121 @@ def test_invalid_status(self):
         self.assertEqual(response.status_code, 422)
         self.assertIn('invalid enrollment record', response.data)
 
 
+class CourseEnrollmentPostTests(BaseCourseEnrollmentTestsMixin, APITestCase):
+    """ Tests for course enrollment POST """
+
+    def request(self, path, data):
+        return self.client.post(path, data, format='json')
+
+    def test_enrollments(self):
+        self.create_program_enrollment('l1')
+        self.create_program_enrollment('l2')
+        self.create_program_enrollment('l3', user=None)
+        self.create_program_enrollment('l4', user=None)
+        post_data = [
+            self.learner_enrollment("l1", "active"),
+            self.learner_enrollment("l2", "inactive"),
+            self.learner_enrollment("l3", "active"),
+            self.learner_enrollment("l4", "inactive"),
+        ]
+        response = self.request(self.default_url, post_data)
+        self.assertEqual(200, response.status_code)
+        self.assertDictEqual(
+            {
+                "l1": "active",
+                "l2": "inactive",
+                "l3": "active",
+                "l4": "inactive",
+            },
+            response.data
+        )
+        self.assert_program_course_enrollment("l1", "active", True)
+        self.assert_program_course_enrollment("l2", "inactive", True)
+        self.assert_program_course_enrollment("l3", "active", False)
+        self.assert_program_course_enrollment("l4", "inactive", False)
+
+    def test_conflict(self):
+        self.create_program_and_course_enrollments('l1')
+        post_data = [self.learner_enrollment("l1")]
+        response = self.request(self.default_url, post_data)
+        self.assertEqual(422, response.status_code)
+        self.assertDictEqual({'l1': CourseStatuses.CONFLICT}, response.data)
+
+    def test_207(self):
+        self.create_program_enrollment('l1')
+        post_data = [self.learner_enrollment("l1"), self.learner_enrollment("l2")]
+        response = self.request(self.default_url, post_data)
+        self.assertEqual(207, response.status_code)
+        self.assertDictEqual(
+            {'l1': CourseStatuses.ACTIVE, 'l2': CourseStatuses.NOT_IN_PROGRAM},
+            response.data
+        )
+
+
+@ddt.ddt
+class CourseEnrollmentPatchTests(BaseCourseEnrollmentTestsMixin, APITestCase):
+    """ Tests for course enrollment PATCH """
+
+    def request(self, path, data):
+        return self.client.patch(path, data, format='json')
+
+    def test_207(self):
+        self.create_program_and_course_enrollments('l1')
+        post_data = [self.learner_enrollment("l1"), self.learner_enrollment("l2")]
+        response = self.request(self.default_url, post_data)
+        self.assertEqual(207, response.status_code)
+        self.assertDictEqual(
+            {'l1': CourseStatuses.ACTIVE, 'l2': CourseStatuses.NOT_IN_PROGRAM},
+            response.data
+        )
+
+    def test_not_found(self):
```

Comment:
Here, too.  I can't tell what exactly isn't found - is it the program, the course, or the status?